zebra-state/src/service/check.rs

```diff
@@ -1,19 +1,31 @@
 //! Consensus critical contextual checks
 
+use std::borrow::Borrow;
+
 use zebra_chain::{
     block::{self, Block},
     parameters::Network,
+    parameters::POW_AVERAGING_WINDOW,
+    work::difficulty::CompactDifficulty,
 };
 
 use crate::{PreparedBlock, ValidateContextError};
 
 use super::check;
 
+pub mod difficulty;
```

Comment:
should this be pub?

Reply:
I'm not sure, let's find out
```suggestion
mod difficulty;
```

Reply:
The `service` module uses `check::difficulty::POW_MEDIAN_BLOCK_SPAN`, so I downgraded it to `pub(crate)`.